include/jwt-cpp/jwt.h

```diff
@@ -2219,15 +2348,15 @@ namespace jwt {
 		 * Get all payload claims
 		 * \return map of claims
 		 */
-		std::unordered_map<typename json_traits::string_type, basic_claim<json_traits>> get_payload_claims() const {
-			return this->payload_claims;
+		std::unordered_map<typename json_traits::string_type, basic_claim_t> get_payload_claims() const {
```

Comment:
Returning map_of_claims would probably look better and cheaper (e.g. by returning a const reference to the member).
However it would be a breaking api change.
We already do have a bunch of breaking changes for 0.4.0 => 0.5.0 so everyone thats not living on master has to check their code anyway and everyone living on master should IMHO expect breaking changes at any time.

But its up to you. Maybe a good middleground would be adding a `get_payload_claims_map()` and deprecating this one. Not having to convert to unordered_map would also remove the std::pair requirement noted on line 2045.

Reply:
There's actually no API breaks, looking at  https://github.com/Thalhammer/jwt-cpp/compare/v0.4.0...master

There are only this change in examples https://github.com/Thalhammer/jwt-cpp/compare/v0.4.0...master#diff-c5ace5b1f9f956b086a7e09a83c2fe4910841987ddc1dc08887a1f5356a83ad1L9 which is from added stream operators

This might be an API break https://github.com/Thalhammer/jwt-cpp/compare/v0.4.0...master#diff-d2b23c35f69c8a9a09ea6dae014a1e5b4b888f6a0d5a6450a8662b7bcf3629d3R83 🤔 

The rest of the changes in examples and tests are new code.

***

I would love to remove these functions! They are only required for the "printing claims example" and provide no functionality for the user... less code less bugs :bug: ?

Right now the `std::pair` is optional, but required for the example.